### PR TITLE
You can no longer disarm wielded weapons

### PR DESCRIPTION
**Pull request by UKNOWH**
## About the PR 
As this has been asked for for quite a while , I sat down and made it. The solution might be a bit hacky , but I can't really think of a better way to do it
**Screenshots**

**Changelog**

:cl:
- change: You can no longer disarm wielded weapons

### CODE REVIEW

**Comment by Snowyypanda** [comment]
welded? you mean wielded?

**Comment by UKNOWH** [comment]
> welded? you mean wielded?

ah yes typo , sorry

**Comment by BlitzGunner225** [comment]
Might be better to have it turn to one hand when disarmed once so it’s safer but not completely safe as these type weapons do a lot of damage

**Comment by UKNOWH** [comment]
> Might be better to have it turn to one hand when disarmed once so it’s safer but not completely safe as these type weapons do a lot of damage

If you can trip them I think it would still drop

File: Content.Server/CombatMode/CombatModeSystem.cs [hunk 1]
```diff
@@ -39,6 +41,18 @@ private void OnEntityActionPerform(EntityUid uid, SharedCombatModeComponent comp
                 return;
 
             var attemptEvent = new DisarmAttemptEvent(args.Target, args.Performer);
+
+            if (EntityManager.TryGetComponent<HandsComponent>(args.Target, out HandsComponent targetHandsComponent)
+                && targetHandsComponent.ActiveHand != null
+                && !targetHandsComponent.ActiveHand.IsEmpty)
+            {
+                Debug.Assert(targetHandsComponent.ActiveHand.HeldEntity != null, "targetHandsComponent.ActiveHand.HeldEntity != null");
```
Comment (mirrorcult) [inline Content.Server/CombatMode/CombatModeSystem.cs:49]: wat, just remove this

File: Content.Server/CombatMode/CombatModeSystem.cs [hunk 2]
```diff
@@ -39,6 +41,18 @@ private void OnEntityActionPerform(EntityUid uid, SharedCombatModeComponent comp
...
+                var inTargetHand = (EntityUid) targetHandsComponent.ActiveHand.HeldEntity;
```
Comment (mirrorcult) [inline Content.Server/CombatMode/CombatModeSystem.cs:50]:
```suggestion
                var inTargetHand = targetHandsComponent.ActiveHand.HeldEntity.Value;
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
approve the idea

File: Content.Server/CombatMode/CombatModeSystem.cs (hunk 1 above)
Comment (UKNOWH) [reply Content.Server/CombatMode/CombatModeSystem.cs:49]: If I don't do an assert , Rider will be mad that held entity could be null

Comment (UKNOWH) [reply Content.Server/CombatMode/CombatModeSystem.cs:49]: Oh nvm I fixed it

File: Content.Server/CombatMode/CombatModeSystem.cs [hunk 3]
```diff
@@ -38,7 +44,21 @@ private void OnEntityActionPerform(EntityUid uid, SharedCombatModeComponent comp
...
+                && !targetHandsComponent.ActiveHand.IsEmpty)
+            {
+                inTargetHand = targetHandsComponent.ActiveHand.HeldEntity!.Value;
+            }
+
+            var attemptEvent = new DisarmAttemptEvent(args.Target, args.Performer,inTargetHand);
+
+            if (inTargetHand != null)
+            {
+                RaiseLocalEvent((EntityUid) inTargetHand, attemptEvent);
```
Comment (mirrorcult) [inline Content.Server/CombatMode/CombatModeSystem.cs:60]:
```suggestion
                RaiseLocalEvent(inTargetHand.Value, attemptEvent);
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Pull request merged**